ibis/backends/duckdb/__init__.py

```diff
@@ -473,6 +473,30 @@ def do_connect(
 
         self.con = duckdb.connect(str(database), config=config, read_only=read_only)
 
+        self._post_connect(extensions)
+
+    @classmethod
+    def from_connection(
+        cls,
+        con: duckdb.DuckDBPyConnection,
+        extensions: Sequence[str] | None = None,
```

Comment:
Is there any reason for us to accept partial arguments like this? What's the issue with every `from_connection` implementation accepting **only** the connection/context/whatever object?

Reply:
My goal was to enable as much of the options you may have when connecting using `connect` as possible. The additional arguments are always optional, though, and should be as though you just passed the `con` object.

PySpark is the one case where I think you have to at least offer `mode`, as you can't infer that in any way from the `SparkSession`.

Reply:
Hm, okay.

Can we remove all optional arguments that aren't strictly required?

Reply:
Longer term we should really try to avoid having arguments like `mode` that we use that can't be determined from the backend itself.

Reply:
Actually, let's leave this as is, but slap `@experimental` on all the `from_connection` implementations.

Reply:
I was looking at it in the meantime. For posterity:

* I think I could remove additional arguments from DuckDB and Exasol
* PySpark and BigQuery have some things that can't be gotten from the connection, as mentioned
* I think SQLite `_type_map` also can't be gotten from an existing connection
* The existing `from_snowpark` supports `create_object_udfs` argument already, so that's kept

> Actually, let's leave this as is, but slap `@experimental` on all the `from_connection` implementations.
Sounds good!

Reply:
Done slapping `@experimental` everywhere.